src/Ocelot/Authorization/ScopesAuthorizer.cs

```diff
@@ -28,14 +28,22 @@ public Response<bool> Authorize(ClaimsPrincipal claimsPrincipal, List<string> ro
                 return new ErrorResponse<bool>(values.Errors);
             }
 
-            var userScopes = values.Data;
+            IList<string> userScopes = values.Data;
 
-            var matchesScopes = routeAllowedScopes.Intersect(userScopes);
+            if (userScopes.Count == 1)

```

Comment:
The `if`-block ought to be relocated to `IClaimsParser` to preserve the existing logic intact. Consequently, you must inject your specialized `IClaimsParser` service to generate the precise list of claims.